src/app/Modal/DeleteWarningModal.tsx

```diff
@@ -0,0 +1,107 @@
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import * as React from 'react';
+import { Modal, ModalVariant, Button, Checkbox, Stack, Split, } from '@patternfly/react-core';
+import {  DeleteAutomatedRules, DeleteWarningType, getFromWarningMap } from './DeleteWarningUtils';
+import { useState } from 'react';
+
+export interface DeleteWarningProps {
+  warningType: DeleteWarningType;
+  items?: Array<string>;
+  visible: boolean;
+  onAccept: () => void;
+  onClose: () => void;
+  setShowDialog: () => void;
+  checkbox?: boolean;
+  setCheckbox?: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const DeleteWarningModal = ({ warningType , items, visible, onAccept, onClose, setShowDialog, checkbox, setCheckbox}: DeleteWarningProps): JSX.Element => {
+  const [doNotAsk, setDoNotAsk] = useState(false);
+  const realWarningType = getFromWarningMap(warningType);
+
+  const description = `${realWarningType?.description}${(typeof items === 'undefined' || items.length <= 1) ?  "":"s"}: [${items?.join(", ")}]. ${realWarningType?.descriptionPartTwo}`
+
+  const onAcceptClose = () => {
+    onAccept();
+    onClose();
+    if (doNotAsk) {
+      setShowDialog();
+    }
+  }
+  
+  return (
+      <Modal
+        title={`Permanently ${realWarningType?.title}?`}
+        description={description}
+        aria-label={realWarningType?.ariaLabel}
+        titleIconVariant="warning"
+        variant={ModalVariant.small}
+        isOpen={visible}
+        showClose
+        onClose={onClose}
+        actions={[  
+          <Stack hasGutter key="modal-footer-stack">
+            <Split key="modal-footer-split">
+              <Button variant="danger" onClick={() => onAcceptClose()}>
+                Delete
+              </Button>
+              <Button variant="link" onClick={onClose}>
+                Cancel
+              </Button>
+            </Split>
+          </Stack> 
+        ]}
+      >
+        <Stack hasGutter key="modal-checkboxes-stack">
+          {(realWarningType === DeleteAutomatedRules) && (typeof setCheckbox !== 'undefined') &&
+          <Checkbox id="clean-rule-enabled" 
+            label="Clean" 
+            description={`Clean will stop any Active Recordings that ${items![0]} created.`}
+            isChecked={checkbox} 
+            onChange={(checked) => setCheckbox(checked)}
```

Comment:
Checking my understanding here:

I wrote this: 
```
  const disableRuleDialog = React.useCallback(() => {
    context.settings.setDeletionDialogsEnabledFor(DeleteWarningType.DeleteAutomatedRules, false);
  }, [context.settings.setDeletionDialogsEnabledFor])
...
...
            <RuleDeleteWarningModal 
            ...                 
            ...
                disableDialog={disableRuleDialog}
            />
```

1. Would there be a point in using useCallback here? The disableDialog function is only invoked when the Delete button is clicked with the checkmark checked, and in that case, we would probably want a re-render anyways, right? 

2. If not then, when doing this
```
            <RuleDeleteWarningModal 
                 ...
                onAccept={() => { handleDelete(rowDeleteData, cleanRuleEnabled); } }
                ....
            />
```
vs 
```
const foo = () => {
     handleDelete(...);
}
...
...
            <RuleDeleteWarningModal 
                 ...
                onAccept={foo}
                ....
            />

```
is there any difference between passing in an anonymous prop function in the first example vs the second?

Reply:
In this particular example, where the "real" function being called ("setDeletionDialogsEnabledFor") doesn't depend on any other piece of state from the component, I don't think it ends up mattering in the end so far as React and rendering goes.

If it did depend on some other piece of state, then I think `useCallback` would be the way to go. This way the function body is properly updated to use fresh state data that it needs. There isn't much reason *not* to use `useCallback` here, other than it being unnecessary and adding a bit of extra boilerplate code. At least this way, if the function definition is changed in the future and begins to depend on some piece of state, it's already nicely set up to do that by just adjusting the dependency array.

If you don't want to use `useCallback`, `const foo = () =>` is better just because it moves code out of the JSX template and into the normal-feeling TypeScript part of the component definition. If you wanted to add some other behaviours to that `onAccept` event handler, it gets really messy doing it the `onAccept={() => { doSomething(); const value = doAComputation(); andSomethingElse(value); } }` way.